components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2237,8 +2238,12 @@ private boolean isIdTokenHintExists(OAuth2Parameters oauth2Params) {
     private boolean isUserAlreadyApproved(OAuth2Parameters oauth2Params, AuthenticatedUser user)
             throws OAuthSystemException {
 
-        return OpenIDConnectUserRPStore.getInstance().hasUserApproved(user, oauth2Params.getApplicationName(),
-                oauth2Params.getClientId());
+        try {
+            return EndpointUtil.isUserAlreadyConsentedForOAuthScopes(user, oauth2Params);
+        } catch (IdentityOAuth2ScopeConsentException | IdentityOAuthAdminException e) {
+            throw new OAuthSystemException("Error occurred while checking scopes are already approved by the user.",
```

Comment:
Shall we improve this error message ? I think we can improve it to make it more understandable.